src/RequestsLibrary/SessionKeywords.py

```diff
@@ -566,12 +565,15 @@ def _common_request(
         method_function = getattr(session, method)
         self._capture_output()
 
+        try:
+            kwargs['cookies']
+        except KeyError:
+            kwargs['cookies'] = self.cookies
+
```

Comment:
You can use `defaultdict` (https://docs.python.org/3/library/collections.html#collections.defaultdict) or `.get()` method with a default value than `try except`

Reply:
Problem with .get is that I need to do this assignment only if cookies is not in the dictionary, and None might be a value... but while I'm writing a solution could be this

> # #
```suggestion
        self._capture_output()

        resp = method_function(
            self._get_url(session, uri),
            params=utils.utf8_urlencode(kwargs.pop('params', None)),
            timeout=self._get_timeout(kwargs.pop('timeout', None)),
            cookies=kwargs.pop('cookies', self.cookies),
            **kwargs)

        log.log_request(resp)
```